lib/kb_orthofinder/kb_orthofinderImpl.py

```diff
@@ -277,31 +278,41 @@ def annotate_plant_transcripts(self, ctx, input):
         if(len(sequences_dict)==0):
             raise Exception("The genome does not contain any protein sequences!")
 
+        # We have a problem with how OrthoFinder arbitrarily replaces 'special' characters in identifiers:
+        # https://github.com/davidemms/OrthoFinder/blob/master/scripts_of/util.py#L181
+        # without 'hacking' the OF code, here I'm trying to anticipate how OF may have changed the identifier
+
+        alt_seq_ids_dict=dict()
+        for og_id in sequences_dict:
+            of_id = og_id.replace(":", "_").replace(",", "_").replace("(", "_").replace(")", "_")
+            alt_seq_ids_dict[of_id]=og_id
+
         output['ftrs'] = len(sequences_dict.keys())
 
         #Create directory for storing new fasta file
-        uuid_string = str(uuid.uuid4())
+        uuid_string = "print_fasta_"+str(uuid.uuid4())
         fasta_file_path=os.path.join(self.scratch,uuid_string)
         os.mkdir(fasta_file_path)
 
         #Reference data is considered immutable but each run modifies results within the directory
         #So here, we copy the reference data directory into scratch
         #The first if condition is for testing purposes
 
-        uuid_string = str(uuid.uuid4())
-        family_file_path=os.path.join(self.scratch,uuid_string,"Reference_Results")
+        uuid_string = "family_data_"+str(uuid.uuid4())
+        family_file_path=os.path.join(self.scratch,uuid_string)
 
         if(self.testing is True):
```

Comment:
Brackets not needed in python; you can also do `if self.testing:` to test boolean conditions.

Reply:
Changed in 9ca841d60089414785b076375ae088febb9d0668